pkg/registry/rbac/escalation_check.go

```diff
@@ -39,3 +44,46 @@ func EscalationAllowed(ctx genericapirequest.Context) bool {
 
 	return false
 }
+
+// BindingAuthorized returns true if the user associated with the context is explicitly authorized to bind the specified roleRef
+func BindingAuthorized(ctx genericapirequest.Context, roleRef rbac.RoleRef, bindingNamespace string, a authorizer.Authorizer) bool {
+	if a == nil {
+		return false
+	}
+
+	user, ok := genericapirequest.UserFrom(ctx)
+	if !ok {
+		return false
+	}
+
+	attrs := authorizer.AttributesRecord{
+		User: user,
+		Verb: "bind",
+		// check against the namespace where the binding is being created.
+		// this allows delegation to bind particular clusterroles within particular namespaces.
+		Namespace:       bindingNamespace,
+		ResourceRequest: true,
+	}
+
+	switch roleRef.Kind {
+	case "ClusterRole":
+		attrs.APIGroup = roleRef.APIGroup
```

Comment:
note about us not having passed validation, but being post-defaulting and conversion, so this is safe since the value won't change again and an invalid value will be rejected by validation